### PR TITLE
UI: TeamsSelectable component should use REST API to build teams hierarchy

### PR DESCRIPTION
**Pull request by vivekratnavel**
### Describe your changes :
- Added an API `/api/v1/teams/hierarchy` to list team hierarchy
- Updated UI code to consume REST API instead of ES for team hierarchy listing

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/1051198/192199954-974f7a7b-dbb8-40f7-bd4c-c8287365eb5a.mov

https://user-images.githubusercontent.com/1051198/192199974-ffc66d7a-fc52-4f92-8097-68cea28d0104.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by chirag-madlani — APPROVED** [review]
Thanks @vivekratnavel It's clear lots of clutter on UI side code.

File: openmetadata-ui/src/main/resources/ui/src/axiosAPIs/teamsAPI.ts [hunk 1]
```diff
@@ -46,6 +47,18 @@ export const getTeams = async (
   return response.data;
 };
 
+export const getTeamsHierarchy = async (isJoinable = false) => {
+  const url = getURLWithQueryFields('/teams/hierarchy');
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/axiosAPIs/teamsAPI.ts:51]: Can we directly use the URL as we don't have any query fields?

File: openmetadata-ui/src/main/resources/ui/src/axiosAPIs/teamsAPI.ts [hunk 2]
```diff
@@ -46,6 +47,18 @@ export const getTeams = async (
...
+
+  const response = await APIClient.get<{ data: TeamHierarchy[] }>(url, {
+    params: {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/axiosAPIs/teamsAPI.ts:54]:
```suggestion
  const response = await APIClient.get<{ data: TeamHierarchy[] }>('/teams/hierarchy', {
    params: {
```

File: openmetadata-ui/src/main/resources/ui/src/components/TeamsSelectable/TeamsSelectable.tsx [hunk 3]
```diff
@@ -41,145 +35,39 @@ const TeamsSelectable = ({
...
       // TODO: Improve type below
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    }).then((res: any) => {
-      const teams: Team[] =
-        res.hits.hits.map((t: { _source: Team }) => t._source) || [];
-      // Build hierarchical tree structure for Tree Select
-      processTeams(teams).then((nodes: Map<string, Node>) => {
-        const nodeValues: Array<Node> = Array.from(nodes.values());
-        setNodes(nodeValues);
+      .then((res: any) => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamsSelectable/TeamsSelectable.tsx:49]: Can we use the proper type here as we have the json schema for this?

File: openmetadata-ui/src/main/resources/ui/src/components/TeamsSelectable/TeamsSelectable.tsx [hunk 4]
```diff
@@ -41,145 +35,39 @@ const TeamsSelectable = ({
...
   useEffect(() => {
     loadOptions();
   }, []);
 
-  const getTreeNodes = (node: Node) => {
-    const teamName = node.team.displayName || node.team.name;
-    const value = node.team.id;
-    const disabled = filterJoinable ? !node.team.isJoinable : false;
+  const getTreeNodes = (team: TeamHierarchy) => {
+    const teamName = team.displayName || team.name;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamsSelectable/TeamsSelectable.tsx:64]:
```suggestion
    const teamName = getEntityName(team);
```

**Comment by vivekratnavel** [comment]
> Thanks @vivekratnavel It's clear lots of clutter on UI side code.

Yes, all of that complexity of building the hierarchy is now moved to the backend side :)

**Comment by vivekratnavel** [comment]
@Sachin-chaurasiya Thank you for the review! I have addressed all of the comments. PTAL

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TeamRepository.java [hunk 5]
```diff
@@ -179,6 +184,107 @@ protected void cleanup(Team team) throws IOException {
...
+      for (TeamHierarchy n : toAdd) team1.getChildren().add(deepCopy(n));
+    }
+
+    return team1;
+  }
+
+  public List<TeamHierarchy> listHierarchy(ListFilter filter, Boolean isJoinable) throws IOException {
+    Fields fields = getFields("parents");
+    Map<UUID, TeamHierarchy> map = new HashMap<>();
+    ResultList<Team> resultList = listAfter(null, fields, filter, 10000, null);
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TeamRepository.java:244]: make 10000 as configurable and pass it as param from queries with default 10000

File: openmetadata-spec/src/main/resources/json/schema/entity/teams/teamHierarchy.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    "displayName": {
+      "description": "Name used for display purposes. Example 'Data Science team'.",
+      "type": "string"
+    },
+    "href": {
+      "description": "Link to the resource corresponding to this entity.",
+      "$ref": "../../type/basic.json#/definitions/href"
+    },
+    "children" : {
```
Comment (harshach) [inline openmetadata-spec/src/main/resources/json/schema/entity/teams/teamHierarchy.json:41]: we can just add this to team.json right, any reason to create another class for this?

**Pull request merged**